### PR TITLE
Add colliderect(), collideswith() to circle

### PR DESCRIPTION
**Pull request by itzpr3d4t0r**
- Added colliderect function
- Added collideswith function
- Added more unittests for circle
- added the pgCollision_RectCircle function

### CODE REVIEW

File: src_c/circle.c [hunk 1]
```diff
@@ -261,6 +261,61 @@ pg_circle_collidepoint(pgCircleObject *self, PyObject *const *args,
...
+            !pg_IntFromObj(args[1], &temp.y) ||
+            !pg_IntFromObj(args[2], &temp.w) ||
+            !pg_IntFromObj(args[3], &temp.h)) {
+            return RAISE(PyExc_TypeError,
+                         "Invalid rect, all 4 fields must be numeric");
+        }
+    }
+    else {
+        return RAISE(PyExc_TypeError,
+                     "Invalid argumets number, must be 1, 2 or 4");
```
Comment (novialriptide) [inline src_c/circle.c:297]: Typo: `argumets` -> `arguments`

File: src_c/circle.c [hunk 2]
```diff
@@ -261,6 +261,61 @@ pg_circle_collidepoint(pgCircleObject *self, PyObject *const *args,
...
+    int result = 0;
+    if (pgCircle_Check(arg)) {
+        result =
+            pgCollision_CircleCircle(&self->circle, &pgCircle_AsCircle(arg));
+    }
+    else if (pgRect_Check(arg)) {
+        result = pgCollision_RectCircle(&(pgRect_AsRect(arg)), &self->circle);
+    }
+    else if (PySequence_Check(arg)) {
+        return pg_circle_collidepoint(self, (PyObject *const *)(&arg), 1);
```
Comment (novialriptide) [inline src_c/circle.c:315]: > `(PyObject *const *)`

This looks strange... @Starbuck5 can you check this?

**Review by novialriptide — CHANGES_REQUESTED** [review]

File: src_c/circle.c (hunk 2 above)
Comment (itzpr3d4t0r) [reply src_c/circle.c:315]: It's just a little hack to not have to add a new function just for this one to work

Comment (itzpr3d4t0r) [reply src_c/circle.c:315]: And it works i've tested it visually

Comment (itzpr3d4t0r) [reply src_c/circle.c]: nvm removed it

**Pull request merged**